sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/SqlTransform.java

```diff
@@ -128,12 +127,12 @@
       sqlEnvBuilder.autoLoadUserDefinedFunctions();
     }
 
-    sqlEnvBuilder.setQueryPlannerClassName(
-        MoreObjects.firstNonNull(
-            queryPlannerClassName(),
-            input.getPipeline().getOptions().as(BeamSqlPipelineOptions.class).getPlannerName()));
-
-    sqlEnvBuilder.setPipelineOptions(input.getPipeline().getOptions());
+    BeamSqlPipelineOptions options =
+        input.getPipeline().getOptions().as(BeamSqlPipelineOptions.class);
+    if (queryPlannerClassName() != null) {
+      options.setPlannerName(queryPlannerClassName());
```

Comment:
This method should not mutate the options.

Reply:
Removed.